src/components/Modal/ViewAllModal/index.tsx

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import { TopicSummary } from "../../../types";
+import Modal from "../index";
+import TextStat from "../../TextStat";
+import percentageIncrease from "../../../util/percentageIncrease";
+import Chip from "../../Chip";
+import Review from "../../Review";
+import IconButton from "../../IconButton";
+import { faFlag, faTrashAlt } from "@fortawesome/free-regular-svg-icons";
+import { faFlag as faFlagged } from "@fortawesome/free-solid-svg-icons";
+import styles from "./style.module.css";
+import dayjs from "dayjs";
+
+interface ViewAllModalProps {
+  show: boolean;
+  onClose: () => void;
+  topic: TopicSummary | undefined;
```

Comment:
The only thing with the `ViewAllModal` is that it will be past a `Topic` instead of `TopicsSummary` & as a result needs to calculate its own status (`Topic` doesn't have the `counts` field) - although if you like I can do this when I hook up the endpoint for it

Reply:
If you could handle this when hooking up the endpoint that would be ideal!